logstreams/src/main/java/io/zeebe/logstreams/impl/log/index/LogBlockIndex.java

```diff
@@ -111,12 +125,24 @@ public synchronized long lookupBlockPosition(final long entryPosition) {
           } else {
             return false;
           }
-        });
+        },
+        indexContext.getKeyInstance(),
+        indexContext.getValueInstance());
 
     return blockPosition.get();
   }
 
-  public synchronized void addBlock(long blockPosition, long blockAddress) {
+  /**
+   * Adds a mapping between a block's position and its address to the log block index.
+   *
+   * @param indexContext the log block index context
+   * @param blockPosition the block's position
+   * @param blockAddress the block's address
+   * @return the position of the block containing the log entry identified by the provided virtual
+   *     position
+   */
+  public void addBlock(
+      final LogBlockIndexContext indexContext, final long blockPosition, final long blockAddress) {
```

Comment:
Is this called by different threads?

Reply:
I think no, otherwise we can get some problems with the `lastVirtualPosition`

Reply:
No, only the writer uses this method. Other than that the lastVirtualPosition is only set once when we restore from a snapshot